Detections/ThreatIntelligenceIndicator/DomainEntity_PaloAlto.yaml

```diff
@@ -17,37 +17,41 @@ triggerThreshold: 0
-    | summarize count() by tostring(tld)
-    | summarize makelist(tld);
-    ThreatIntelligenceIndicator
-    | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
-    | where Active == true
-    // Picking up only IOC's that contain the entities we want
-    | where isnotempty(DomainName)
-    | join (
-        CommonSecurityLog
-        | where TimeGenerated > ago(dt_lookBack)
-        | where DeviceVendor == 'Palo Alto Networks'
-        | where DeviceEventClassID == 'url'
-        | where DeviceVendor == 'PaloAltoNetworks'
-        //Extract domain from field
-        | extend domain = extract("(([a-z0-9]+(-[a-z0-9]+)*\\.)+[a-z]{2,})", 1, tolower(AdditionalExtensions))
-        | where isnotempty(domain)
-        | extend parts = split(domain, '.')
-        //Split out the TLD
+        | where TimeGenerated > ago(ioc_lookBack)
+        | where isnotempty(DomainName)
+        | extend parts = split(DomainName, '.')
         | extend tld = parts[(array_length(parts)-1)]
-        //Validate parsed domain by checking TLD against lof of TLDs from threat feed
-        | where tld in~ (list_tlds)
-        | extend CSL_TimeGenerated = TimeGenerated
-    ) on $left.DomainName==$right.domain
-    | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
-    | project LatestIndicatorTime, Description, ActivityGroupNames, IndicatorId, ThreatType, ExpirationDateTime, ConfidenceScore, CSL_TimeGenerated, DeviceAction, DestinationIP, DestinationPort, DeviceName, SourceIP, SourcePort, ApplicationProtocol, RequestMethod
-    | extend timestamp = CSL_TimeGenerated, IPCustomEntity = SourceIP, HostCustomEntity = DeviceName
+        | summarize count() by tostring(tld)
+        | summarize makelist(tld);
+        ThreatIntelligenceIndicator
+        | where TimeGenerated >= ago(ioc_lookBack) and ExpirationDateTime > now()
+        | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+        | where Active == true
+        // Picking up only IOC's that contain the entities we want
+        | where isnotempty(DomainName)
+        | join (
+            CommonSecurityLog
+            | where TimeGenerated > ago(dt_lookBack)
+            | where DeviceVendor == 'Palo Alto Networks'
+            | where DeviceEventClassID == 'url'
+            //Uncomment the line below to only alert on allowed connections
+            //| where DeviceAction != "block-url"
+            //Extract domain from RequestURL, if not present extarct it from AdditionalExtentions
+            | extend Url = columnifexists("RequestURL", "None")
+            | extend Url = iif(Url == "" and AdditionalExtensions !startswith "PanOS", extract("([^\"]+)", 1, tolower(AdditionalExtensions)), trim('"', Url))
+            | extend Url = iif(Url !startswith "http://" and ApplicationProtocol != "ssl", strcat('http://', Url), iif(Url !startswith "https://" and ApplicationProtocol == "ssl", strcat('https://', Url), Url))

```

Comment:
recommend we use =~ on the ApplicationProtocol == "ssl", just in case the name case changes